### PR TITLE
Add architecture and best-practices sections

### PR DESCRIPTION
**Pull request by djmitche**
@selenamarie @hwine what do you think?

### CODE REVIEW

File: architecture/audience.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Audience
+========
+
+The audience for this chapter on architecture is release engineering contributors (including employees, of course).
```
Comment (selenamarie) [inline architecture/audience.rst]: Pedantic level: 11 -- maybe put `(releng)` in here as we stop referring to 'release engineering' after this point.

**Comment by selenamarie** [comment]
r+ yay!

File: index.rst [hunk 2]
```diff
@@ -25,8 +25,9 @@ Contents:
    schedulerdb.rst
    statusdb.rst
    releng101/index.rst
-   coding.rst
    jobs_not_starting.rst
+   best-practices/index
+   architecture/index
```
Comment (hwine) [inline index.rst]: nit: everything else has .rst extension, I vote for being consistent, but not a blocker.

**Comment by hwine** [comment]
r+ as well

**Pull request merged**